src/components/Message/Message.tsx

```diff
@@ -609,8 +609,7 @@ const MessageWithContext = <
   const onOpenThread = () => {
     if (onThreadSelect) {
       onThreadSelect(message);
-    }
-    if (openThread) {
+    } else if (openThread) {
       openThread(message);
```

Comment:
If `onThreadSelect` is provided, which means user is managing thread screen separately on app level. So we shouldn't set thread on current channel.

Reply:
I don't think we can or should assume this; and this will break a lot of other things related to input.

Reply:
umm ok ... closing this for now. lets pick it up later. 

Its just that, the fix we have made in our SampleApp looks a bit odd to me. We have to set and clear thread state on ChannelScreen, when I don't have any thread related logic on ChannelScreen.

Reply:
And if someone doesn't add that fix, it causes a issue

- Open channel
- Go to some thread
- come back to channel
- send a message

message here will be sent to thread instead of original channel.

Reply:
Yeah let's look at it I think there is a way to do it easier; the issue now is that thread is passed via nav instead of in context. The `Channel` still thinks it's the relevant screen so it will be updating like it's live. E.G. upload an image, start to pull back the nav on ios, the image will be there too, and actually will have uploaded 2x.